practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/repository/HapiFhirRepository.java

```diff
@@ -82,6 +86,37 @@ public String savePractitionerRole(PractitionerRole practitionerRole) {
         return responseElement.getIdElement().getIdPart();
     }
 
+    @Override
+    public List<PractitionerRole> getPractitionerRolesByIdentifier(String identifier) {
+        // Log the request
+        if (logger.isInfoEnabled()) {
+            logger.info(FhirRepo.GET_PRACTITIONER_ROLES_BY_IDENTIFIER.message() + identifier);
+        }
+
+        List<PractitionerRole> practitionerRoles = new ArrayList<>();
+
+        var fullTypeName = PractitionerRole.class.getTypeName();
+        var lastNotationIndex = fullTypeName.lastIndexOf('.');
+        var typeName = fullTypeName.substring(lastNotationIndex + 1);
+
+        var results = fhirContextWrapper.searchResourceWithInclude(
+                typeName,
+                new Include("Practitioner:identifier"),
+                PractitionerRole.PRACTITIONER.hasChainedProperty(
+                        Practitioner.IDENTIFIER.exactly().identifier(identifier)));
```

Comment:
I recommend checking what [ARTS-185] is putting in this field - it is likely to include system and code, so not sure how that needs to be search for.